packages/services/src/contents/index.ts

```diff
@@ -1571,3 +1582,235 @@ namespace Private {
     return extension;
   }
 }
+
+class ContentProviderRegistry implements IContentProviderRegistry {
+  register(provider: IContentProvider): void {
+    this._providers.push(provider);
+  }
+
+  getProvider(filePath: string): IContentProvider | undefined {
```

Comment:
Done in ad6ae159c2535c05a274f19dc17463d8080f9c36. This require to define a default provider explicitly, otherwise the list could be empty if one was not registered.